segmentation/include/pcl/segmentation/extract_clusters.h

```diff
@@ -41,10 +41,148 @@
 
 #include <pcl/pcl_base.h>
 
+#include <pcl/common/utils.h>
 #include <pcl/search/pcl_search.h>
 
 namespace pcl
 {
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/** \brief Decompose a region of space into clusters based on the Euclidean distance between points
+  * \param it cloud iterator for iterating over the points
+  * \param cloud the point cloud message
+  * \param additional_filter_criteria functor which is used as an additonal criteria that all points being added to the cluster must satisfy
+  * \param tree the spatial locator (e.g., kd-tree) used for nearest neighbors searching
+  * \note the tree has to be created as a spatial locator on \a cloud and \a indices
+  * \param tolerance the spatial cluster tolerance as a measure in L2 Euclidean space
+  * \param clusters the resultant clusters containing point indices (as a vector of PointIndices)
+  * \param min_pts_per_cluster minimum number of points that a cluster may contain (default: 1)
+  * \param max_pts_per_cluster maximum number of points that a cluster may contain (default: max int)
+  * \warning It is assumed that that the tree built on the same indices and cloud as passed here,
+  * if that is not the case then things can go very wrong. For speed reasons, we only check the sizes and not the content
+  * \ingroup segmentation
+  */
+  template <typename PointT, typename FunctorT> void
+   extractEuclideanClusters(
+      ConstCloudIterator<PointT> &it, const PointCloud<PointT> &cloud, FunctorT additional_filter_criteria,
```

Comment:
Should we take the iterator by value or mutable reference? @SergioRAgostinho I don't now how easily copyable the iterator is

Reply:
It looks like a simple pointer copy. 
https://github.com/PointCloudLibrary/pcl/blob/009d73dbb51fcea8dbf8daeb1229dbb48e0f3870/common/include/pcl/cloud_iterator.h#L116-L186

Reply:
Let's stick to a reference in that case